networkx/drawing/nx_pylab.py

```diff
@@ -470,6 +476,40 @@ def draw_networkx_nodes(
         labelleft=False,
     )
 
+    if auto_margins:
+        try:  # sets x and y margins separately if node_size is iterable
+            node_coords_sizes = list(zip(pos.items(), node_size))
+
+            x_min = sorted(node_coords_sizes, key=lambda a: a[0][1][0], reverse=False)[
+                0
+            ]
+            x_max = sorted(node_coords_sizes, key=lambda a: a[0][1][0], reverse=True)[0]
```

Comment:
```suggestion
            sortx = sorted(node_coords_sizes, key=lambda a: a[0][1][0])
            x_min = sortx[0]
            x_max = sortx[-1]
```
Saves sorting twice if you use first and last result of sort.
It might be better to use `min` and `max` than sort -- not sure.
The same suggestion applies to `y_min` and `y_max`.